src/GafferScene/SceneWriter.cpp

```diff
@@ -189,6 +189,27 @@ void SceneWriter::writeLocation( const GafferScene::ScenePlug *scene, const Scen
 		output->writeTransform( new IECore::M44dData( transform ), time );
 	}
 
+	// query the set membership for this location and write as tags.
+	const std::vector<InternedString>& setNames = scene->setNamesPlug()->getValue()->readable();
+
+	SceneInterface::NameList locationSets;
+	locationSets.reserve( setNames.size() );
+
+	for( size_t i = 0, numSets = setNames.size(); i < numSets; ++i )
+	{
+		context->set( ScenePlug::setNameContextName, setNames[i] );
+		const PathMatcher& pathMatcher = scene->setPlug()->getValue()->readable();
+
+		if( pathMatcher.match( scenePath ) == Filter::ExactMatch )
```

Comment:
Shouldn't this be `| Filter::ExactMatch` rather than `== Filter::ExactMatch`? Otherwise it won't write out the set membership if there are ancestors or descendants in the set. If that's the case, could you update the test so that it's capable of catching this error and make the fix?